modules/msdfgen/SCsub

```diff
@@ -36,7 +36,13 @@ if env["builtin_msdfgen"]:
     ]
     thirdparty_sources = [thirdparty_dir + file for file in thirdparty_sources]
 
-    env_msdfgen.Prepend(CPPPATH=["#thirdparty/freetype/include", "#thirdparty/msdfgen", "#thirdparty/nanosvg"])
+    # Treat msdfgen headers as system headers to avoid raising warnings. Not supported on MSVC.
+    if not env.msvc:
+        env_msdfgen.Append(CPPFLAGS=["-isystem", Dir("#thirdparty/msdfgen").path])
+    else:
+        env_msdfgen.Prepend(CPPPATH=["#thirdparty/msdfgen"])
+
+    env_msdfgen.Prepend(CPPPATH=["#thirdparty/freetype/include", "#thirdparty/nanosvg"])
```

Comment:
This change shouldn't be needed as warnings are disabled on the whole `env_msdfgen` (see line 13).

Though in theory that's not how it should be, there should be an `env_thirdparty = env_msdfgen.Clone()` on which warnings are disabled, used to compile the thirdparty code. But the Godot source files added to `env_msdfgen` should still use warnings (and would then need this trick if they include msdfgen headers that trigger warnings).

Reply:
That's surprising, as this code is compiling with `-w` so it shouldn't raise any warnings.

Reply:
It appears that doing `-isystem` and then including the header explicitly causes something, not sure, declaring it in all possible paths seems to be required

Reply:
The problem is raised in `text_server_adv` to clarify

Reply:
Then I will test that too, thank you

Reply:
> you seemed to mention it but then removed the comment

Yes sorry got confused by my own logic for a moment